trieste/acquisition/function.py

```diff
@@ -201,6 +203,79 @@ def update_acquisition_function(
         return function
 
 
+class DeepGaussianProcessExpectedImprovement(SingleModelAcquisitionBuilder):
```

Comment:
I would make this more general, not tied specifically to `DeepGaussianProcess` class of models - this can be easily done by providing a sampler as an input and the `ProbabilisticModel` has to have a method that provides `mean_samples`

Reply:
I've had a go at this now